### PR TITLE
Remove mockpool based tests

### LINKED ISSUE #4692: Remove mockpool based tests
### Problem

The `MockPool.js` mocks the database layer in the REST API so it can capture SQL queries, parse them, and use them to assert in tests. Executing REST API calls to verify SQL syntax is a violation of separation of concerns. Frequently the mockpool based tests also have flaky tests when ran via CI.

### Solution

* Replace any tests that use the `sqlQuery` object returned in the REST API response when used with MockPool to add coverage to existing/new specs or extract SQL verification to separate unit tests by verifying query generation into separate, testable `getQuery` methods on the service
* Remove `MockPool.js`

### Alternatives

_No response_

### LINKED ISSUE #7656: REST API tests can sometimes hang indefinitely
### Description

REST API tests have started to freeze and never recover more latey, resulting in workflow timeouts in CI. https://github.com/hashgraph/hedera-mirror-node/actions/runs/7732600842/job/21082882702

### Steps to reproduce

./gradlew :rest:build

### Additional context

_No response_

### Hedera network

other

### Version

main

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

Removes mockpool based tests.

- Fix possible test case resource leak caused by incorrect sinon fake timers config
- Fix summary test reporter threshold
- Remove mockpool and supporting functions
- Remove mockpool based tests cases from accounts and balances
- Upgrade to codecov-action@v4

**Related issue(s)**:

Fixes #4692
Fixes #7656

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/jest.config.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ const config = {
   globalSetup: './__tests__/globalSetup.js',
   globalTeardown: './__tests__/globalTeardown.js',
   maxWorkers,
-  reporters: [['github-actions', {silent: false}], 'jest-junit', ['summary', {summaryThreshold: 1}]],
+  reporters: [['github-actions', {silent: false}], 'jest-junit', ['summary', {summaryThreshold: 0}]],
```
Comment (xin-hedera) [inline hedera-mirror-rest/jest.config.js:31]: summary threshold logic,

if the number of test suites executed > threshold and there are failures, then
  show the failed test cases at the end
endif

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -59,5 +59,7 @@ jobs:
           arguments: :${{matrix.project}}:build --scan ${{ secrets.GRADLE_ARGS }}
 
       - name: Upload coverage report
-        uses: codecov/codecov-action@v3
-        if: always() && matrix.schema == 'v1'
+        uses: codecov/codecov-action@v4
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        if: always() && matrix.schema == 'v1' && matrix.project != 'rest:monitoring' && matrix.project != 'test'
```
Comment (xin-hedera) [inline .github/workflows/gradle.yml:65]: codedov-action v4 complains about no coverage info available for rest:monitoring and test (the acceptance test)

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by jascks — APPROVED** [review]
Spent too much of my "childhood" in here. Glad to see it's gone! LGTM!

**Pull request merged**